build-tools/create-packs/ConfigureLocalWorkload.targets

```diff
@@ -0,0 +1,124 @@
+    <_TemplatesInputs Include="$(XamarinAndroidSourcePath)src\Microsoft.Android.Templates\**" />
+    <_TemplatesOutputs Include="$(BuildOutputDirectory)lib\template-packs\microsoft.android.templates.$(AndroidPackVersion).nupkg" />
+  </ItemGroup>
+
+  <Target Name="CreateLocalRuntimeLists"
+      Inputs="$(MSBuildThisFile);@(_RuntimeListInputs)"
+      Outputs="@(_RuntimeListOutputs)">
+    <MSBuild
+        Projects="$(MSBuildThisFileDirectory)Microsoft.Android.Runtime.proj"
+        Properties="FrameworkListFile=%(_RuntimeListOutputs.Identity);AndroidRID=%(_RuntimeListOutputs.AndroidRID)"
+        Targets="_GetRuntimePackItems;_GenerateFrameworkListFile"
+    />
+  </Target>
+
+  <Target Name="CreateLocalFrameworkLists"
+      Inputs="$(MSBuildThisFile);@(_FrameworkListInputs)"
+      Outputs="@(_FrameworkListOutputs)">
+    <MSBuild
+        Projects="$(MSBuildThisFileDirectory)Microsoft.Android.Ref.proj"
+        Properties="FrameworkListFile=%(_FrameworkListOutputs.Identity)"
+        Targets="_GetTargetingPackItems;_GenerateFrameworkListFile"
+    />
+  </Target>
+
+  <Target Name="PackLocalTemplates"
+      Inputs="@(_TemplatesInputs)"
+      Outputs="@(_TemplatesOutputs)">
+    <ItemGroup>
+      <_PackProps Include="-v:n -c $(Configuration)" />
+      <_PackProps Include="-p:IncludeSymbols=False" />
+      <_PackProps Include="-p:OutputPath=$(BuildOutputDirectory)lib\template-packs" />
+      <_PackProps Include="-p:TemplatePackVersion=$(AndroidPackVersion)" />
+      <_PackProps Include="-p:PackageId=microsoft.android.templates" />
+    </ItemGroup>
+    <Exec Command="&quot;$(DotNetPreviewTool)&quot; pack @(_PackProps, ' ') &quot;$(XamarinAndroidSourcePath)src\Microsoft.Android.Templates\Microsoft.Android.Templates.csproj&quot;" />
+  </Target>
+
+  <UsingTask TaskName="GetAndroidWorkloadExtends" TaskFactory="RoslynCodeTaskFactory" AssemblyFile="$(MSBuildToolsPath)\Microsoft.Build.Tasks.Core.dll">
+    <ParameterGroup>
+      <JsonFilePath ParameterType="System.String" Required="true" />
+      <ExtendsElement Output="true" ParameterType="System.String" />
+    </ParameterGroup>
+    <Task>
+      <Using Namespace="System" />
+      <Using Namespace="System.IO" />
+      <Code Type="Fragment" Language="cs">
+        <![CDATA[
+        var jsonContent = File.ReadAllText(JsonFilePath);
+        var startElement = "\"extends\" : [";
+        var startIndex = jsonContent.IndexOf(startElement);
+        if (startIndex != -1) {
+            startIndex += startElement.Length;
+            var endIndex = jsonContent.IndexOf("]", startIndex);
+            if (endIndex != -1) {
+                ExtendsElement = jsonContent.Substring(startIndex, endIndex - startIndex)?.Trim();
+            }
+        }
```

Comment:
Should we maybe `LogError()` in the `else` block for both of these? I imagine some change in the `.json` could silently fail.

Reply:
Sounds good, I figured we would fail during the android-deps workload install in that case but better to error earlier